### PR TITLE
=str #16866 fail fold and foreach future when function throws

### PR DESCRIPTION
**Pull request by patriknw**
Refs #16866

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala [hunk 1]
```diff
@@ -154,6 +156,12 @@ object Sink extends SinkApply {
           promise.success(())
           ctx.finish()
         }
+
+        override def decide(cause: Throwable): Supervision.Directive = {
+          // supervision will be implemented by #16916
+          promise.tryFailure(cause)
+          super.decide(cause)
```
Comment (bantonsson) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala]: I guess that means that #16916 will allow the supervision to decide if the promise is completed with the failure?

**Comment by bantonsson** [comment]
LGTM

File: akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala (hunk 1 above)
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala]: exactly, I'll add a note to the ticket

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
Refs #16866

**Pull request merged**